### PR TITLE
feat: support slash inline command

### PR DESCRIPTION
**Pull request by duongtn811**
## Issues
- Support inline slash

## Loom video

https://www.loom.com/share/23c753dbad4f4384b5f7f1ea3b86d848

### CODE REVIEW

**Review by damianrodrigues — CHANGES_REQUESTED** [review]
https://www.loom.com/share/c1125298cdec4fac96d4bb6048d06372

**Comment by duongtn811** [comment]
Hi @hoantran-it @damianrodrigues 
I have fixed the comments. could you help review it again?

**Comment by duongtn811** [comment]
The loom to explain my fix
https://www.loom.com/share/ca2b2ddf6463494982a6c97f3ce0fb9c

File: src/plugins/BlockMenuTrigger.tsx [hunk 1]
```diff
@@ -42,6 +44,19 @@ function isFolded(state, node) {
   return false;
 }
 
+function checkMatch(state, match) {
+  const currentText = state.selection.$from.parent.content?.content?.[0].text;
```
Comment (damianrodrigues) [inline src/plugins/BlockMenuTrigger.tsx:48]: Can we clarify what is [0]? Either you can use array destructuring and give a descriptive variable name or put a comment explaining what this is

File: src/plugins/BlockMenuTrigger.tsx [hunk 2]
```diff
@@ -115,10 +140,23 @@ export default class BlockMenuTrigger extends Extension {
...
+                  return match ? true : null;
+                }) ||
+                run(view, pos, pos, OPEN_INLINE_REGEX, (state, match) => {
+                  if (!this.options.onCheckBlockMenuState()) {
+                    return false;
+                  }
+
+                  return match ? true : null;
+                })
+              );
```
Comment (damianrodrigues) [inline src/plugins/BlockMenuTrigger.tsx:159]: Same comment as the above

File: src/plugins/BlockMenuTrigger.tsx [hunk 3]
```diff
@@ -94,14 +109,24 @@ export default class BlockMenuTrigger extends Extension {
...
+                  }) ||
+                  run(view, pos, pos, OPEN_INLINE_REGEX, (state, match) => {
+                    if (match) {
+                      this.options.onOpen(match[3]);
+                    } else {
+                      this.options.onClose();
+                    }
+                    return null;
+                  })
+                );
```
Comment (damianrodrigues) [inline src/plugins/BlockMenuTrigger.tsx:129]: This part is a bit confusing, we are expecting to run the function and return null or then run another function and also can return null? Would it be possible to unify this into one function?

File: src/plugins/BlockMenuTrigger.tsx [hunk 4]
```diff
@@ -94,14 +109,24 @@ export default class BlockMenuTrigger extends Extension {
...
+                    if (match) {
+                      this.options.onOpen(match[1]);
+                    } else {
+                      this.options.onClose();
+                    }
+                    return null;
+                  }) ||
+                  run(view, pos, pos, OPEN_INLINE_REGEX, (state, match) => {
+                    if (match) {
+                      this.options.onOpen(match[3]);
```
Comment (damianrodrigues) [inline src/plugins/BlockMenuTrigger.tsx:123]: Let's use constants to clarify what's the number `3` here

File: src/components/BlockMenu.tsx [hunk 5]
```diff
@@ -303,25 +312,63 @@ class BlockMenu extends React.Component<Props, State> {
...
 
     if (parent) {
       dispatch(state.tr.insertText("", parent.pos, state.selection.to));
     }
   }
 
+  insertSelection() {
+    const { state, dispatch } = this.props.view;
+
+    const offset = state.tr.selection.anchor + 1;
```
Comment (damianrodrigues) [inline src/components/BlockMenu.tsx:343]: What's `1` here, let's use a constant or put a comment explaining what's this

File: src/components/BlockMenu.tsx [hunk 6]
```diff
@@ -303,25 +312,63 @@ class BlockMenu extends React.Component<Props, State> {
...
+  removeSlashCharacter() {
+    const { state, dispatch } = this.props.view;
+
+    const parent = findParentNode(node => !!node)(state.selection);
+    const lastSlashIndex = parent?.node.textContent.lastIndexOf("/");
+
+    dispatch(
+      state.tr.insertText(
+        "",
+        parent.pos + lastSlashIndex + 1,
```
Comment (damianrodrigues) [inline src/components/BlockMenu.tsx:324]: What's `1`? Let's use a constant or a comment explaining this

**Review by damianrodrigues — CHANGES_REQUESTED** [review]
Amazing work, just a few nits

File: src/plugins/BlockMenuTrigger.tsx (hunk 1 above)
Comment (duongtn811) [reply src/plugins/BlockMenuTrigger.tsx:48]: Screen Shot 2022-04-19 at 09 47 48

actually, it's the structure of a content inside a node. I am not sure why there always is one TextContent but it's an array

**Comment by duongtn811** [comment]
Hello @damianrodrigues 
I have refactored to have better code and also comment on some points to explain my work. Could you please help take a look again?

**Review by damianrodrigues — APPROVED** [review]
LGTM!

**Pull request merged**